### DISCUSSION TITLE
Database Connection Slots full. between Saleor deployment and DB

### DISCUSSION THREAD

**Discussion by KrEnjuTSU** [saleor/saleor#8173] [Deployment]
Hello everyone,

I'm getting this error "'FATAL:  remaining connection slots are reserved for non-replication superuser connections'" when I'm trying to make API calls using the graph ql engine. when I check in the digital ocean(where I have deployed the DB) I'm having 45 active connections. Can help me by giving some insights, is there something wrong with the deployment?

**Comment by karolkielecki** [comment]
@KrEnjuTSU Hi, maybe this thread on the Digital Ocean forum will help you out: https://www.digitalocean.com/community/questions/fatal-remaining-connection-slots-are-reserved-for-non-replication-superuser-connections

**Reply by ohuu** [reply]
I'm hitting exactly the same problem. I'm using Postgres in Digital Ocean and when I look into `pg_stat_activity` I can see loads of idle connections. These idle connections are stuck running simple select queries which never terminate.

For example:

```
datid            | xxxxx
datname          | xxxx
pid              | xxxxx
leader_pid       |
usesysid         | xxxx
usename          | xxxx
application_name |
client_addr      | xxxxx
client_hostname  |
client_port      | xxxxx
backend_start    | 2022-07-04 10:30:07.684888+00
xact_start       |
query_start      | 2022-07-04 10:30:07.737256+00
state_change     | 2022-07-04 10:30:07.739338+00
wait_event_type  | Client
wait_event       | ClientRead
state            | idle
backend_xid      |
backend_xmin     |
query_id         | -7027771156210260856
query            | SELECT "channel_channel"."id", "channel_channel"."name", "channel_channel"."is_active", "channel_channel"."slug", "channel_channel"."currency_code", "channel_channel"."default_country", EXISTS(SELECT (1) AS "a" FROM "order_order" U0 WHERE U0."channel_id" = "channel_channel"."id" LIMIT 1) AS "has_orders" FROM "channel_channel" WHERE "channel_channel"."id" IN (1)
backend_type     | client backend
```

I don't think the solution is simply increasing the number of connections (I have 47 available) because these idle connections will eventually use up all available connections. The problem is that Saleor is running queries that for some reason are getting blocked.
